src/MPQEditor/MPQEditor.ts

```diff
@@ -0,0 +1,528 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as cp from "child_process";
+import * as fs from "fs";
+import * as path from "path";
+import * as vscode from "vscode";
+
+import { Balloon } from "../Utils/Balloon";
+import { Logger } from "../Utils/Logger";
+import { getNonce } from "../Utils/external/Nonce";
+import { getUri } from "../Utils/external/Uri";
+import { MPQData } from "./MPQData";
+
+export class MPQEditorProvider implements vscode.CustomTextEditorProvider {
+  public static readonly viewType = "one.editor.mpq";
+  public static readonly fileExtension = ".mpq.json";
+
+  private _disposables: vscode.Disposable[] = [];
+  private _mpqDataMap: any = {};
+
+  /**
+   * register commands of MPQEditorProvider
+   */
+  public static register(context: vscode.ExtensionContext): void {
+    const provider = new MPQEditorProvider(context);
+    const registrations = [
+      vscode.window.registerCustomEditorProvider(
+        MPQEditorProvider.viewType,
+        provider,
+        {
+          webviewOptions: { retainContextWhenHidden: true },
+        }
+      ),
+      // Add command registration here
+      vscode.commands.registerCommand(
+        "one.editor.mpq.showFromDefaultExplorer",
+        (uri) => {
+          MPQEditorProvider.createMPQJson(uri);
+        }
+      ),
+    ];
+
+    registrations.forEach((disposable) =>
+      context.subscriptions.push(disposable)
+    );
+  }
+
+  /**
+   * find unoccupied mpq file-name
+   */
+  public static findMPQName(baseMPQName: string, dirPath: string): string {
+    if (baseMPQName.length === 0) {
+      throw new Error("Invalid mixed precision quantization file name");
+    }
+    const baseName = baseMPQName;
+    let mpqName: string = "";
+    for (let i = 0; ; i++) {
```

Comment:
How about putting a large number as a limit for safety?

Reply:
> how about putting a large number as a limit for safety in the for loop?

 @dayo09
Thank you. Yes. It seems like an option, but which one? 100? 1000? ...

Reply:
@dayo09 
Other option: 
we may query `fs` how many `mpq.json` files are there (`max_num`), and set maximal number of iterations as `max_num+1`.